hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,97 @@
+defmodule Game do
+  @moduledoc false
+
+  defmodule State do
+    defstruct game_state: :initializing, turns_left: 7, letters: [], used: [], last_guess: "", random_word: ""
+  end
+
+  @doc "Create new game with default state and random word"
+  def newGame do
+    %State{ random_word: Dictionary.WordList.random_word() }
+  end
+
+  @doc "Create new game state for unit test"
+  def newGame( word ) do
+    %State{ random_word: word }
+  end
+
+  @doc "Returns the tally for the given game"
+  def game_tally( state ) do
+    %{ game_state: state.game_state,
+      turns_left: state.turns_left,
+      letters: word_masking_for_user(state) |> word_split(),
+      used: state.used,
+      last_guess: state.last_guess }
+  end
+
+  @doc "returns a tuple containing the updated game state and a tally"
+  def make_move( game, guess_char ) do
+
+    guess_char = String.downcase(guess_char) |> String.trim
+    game =  handle_guess_answer(
```

Comment:
## -5 bug

Did you notice that your last test looked like this:

~~~ elixir
  test "_won - Game with random word cat assert :won state " do
    game = Hangman.new_game("cat")
    tally1  = Hangman.tally(game)
    # 1 Move
    {game, tally1} = Hangman.make_move(game,"c")
    # 2 Move
    {game, tally1} = Hangman.make_move(game,"a")
    # 3 Move
    {game, tally1} = Hangman.make_move(game,"t")
    {game, tally1} = Hangman.make_move(game,"K")
    assert tally1.game_state == :won
  end
~~~

Why did you need the extra guess at the end? Because without it you don't get a :won state. That's because the code here checks to see if the word so far equals the word to guess _before_ to add the current guess into it.